compiler/rustc_passes/src/check_attr.rs

```diff
@@ -400,6 +402,81 @@ impl CheckAttrVisitor<'tcx> {
         }
     }
 
+    /// Checks if `#[rustc_args_required_const]` is applied to a function and has a valid argument.
+    fn check_rustc_args_required_const(
+        &self,
+        attr: &Attribute,
+        span: &Span,
+        target: Target,
+        item: Option<&Item<'_>>,
+    ) -> bool {
+        match target {
+            Target::Fn | Target::Method(..) | Target::ForeignFn => {
+                let mut invalid_args = vec![];
+                if let Some(meta_item_list) = attr.meta_item_list() {
+                    for meta in meta_item_list {
+                        if let Some(lit) = meta.literal() {
+                            match lit.kind {
+                                LitKind::Int(val, _) => {
+                                    if let Some(item) = item {
+                                        match &item.kind {
+                                            ItemKind::Fn(sig, ..) => {
+                                                if val >= sig.decl.inputs.len() as u128 {
```

Comment:
this is a really deep indentation,

Can you convert some branches into early exists instead?

r=me after that